### PR TITLE
Open Distro for Elasticsearch deployment

### PR DESCRIPTION
**Pull request by manuasir**
Hello,

This PR implements the roles for a fully secured Open Distro for Elasticsearch deployment. This is an adapted and updated version of the resources found in the official [Open Distro](https://opendistro.github.io/for-elasticsearch-docs/docs/other/) resources for Ansible. Related issue #185 .
- Ability to deploy as many cluster nodes as required.
- Security plugin: Certificate generation
- Internal users configuration

Regards

### CODE REVIEW

File: roles/opendistro/opendistro-elasticsearch/tasks/SecurityActions.yml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    when: install.changed
+
+  - name: Set the Admin user password
+    shell: >
+      sed -i 's,{{ opendistro_admin_password }},'$(sh {{ opendistro_sec_plugin_tools_path }}/hash.sh -p {{ opendistro_admin_password }} | tail -1)','
+      {{ opendistro_sec_plugin_conf_path }}/internal_users.yml
+    run_once: true
+    when: install.changed
+
+  - name: Set the kibanaserver user pasword
```
Comment (jm404) [inline roles/opendistro/opendistro-elasticsearch/tasks/SecurityActions.yml:68]: `kibanaserver  role/user` sounds better for me

File: roles/opendistro/opendistro-elasticsearch/handlers/main.yml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+---
+- name: restart elasticsearch
+  service: name=elasticsearch state=restarted
```
[the remark this answers is not in the document]
Comment (manuasir) [reply roles/opendistro/opendistro-elasticsearch/handlers/main.yml:3]: Done here 8fd6849

File: roles/opendistro/opendistro-elasticsearch/tasks/LocalActions.yml [hunk 3]
```diff
@@ -0,0 +1,51 @@
+---
+- block:
+
+  - name: Local action | Create local temporary directory for certificates generation
+    local_action:
+      module: file
+      path: /tmp/opendistro-nodecerts
```
[the remark this answers is not in the document]
Comment (manuasir) [reply roles/opendistro/opendistro-elasticsearch/tasks/LocalActions.yml:7]: Done here a2fe757.

File: roles/opendistro/opendistro-elasticsearch/tasks/LocalActions.yml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+      state: directory
+    run_once: true
+    
+  - name: Local action | Download certificates generation tool
+    local_action:
+      module: get_url
+      url: https://releases.floragunn.com/search-guard-tlstool/1.7/search-guard-tlstool-1.7.zip
```
[the remark this answers is not in the document]
Comment (manuasir) [reply roles/opendistro/opendistro-elasticsearch/tasks/LocalActions.yml:14]: Done here a2fe757.

File: roles/opendistro/opendistro-elasticsearch/tasks/SecurityActions.yml [hunk 5]
```diff
@@ -26,14 +26,14 @@
       - admin.key
       - admin.pem
 
-  - name: Copy the opendistro security configuration file to cluster
+  - name: Copy the OpenDistrosecurity configuration file to cluster
```
Comment (jm404) [inline roles/opendistro/opendistro-elasticsearch/tasks/SecurityActions.yml:29]: OpenDistro security

**Pull request merged**